File: java/1.7/factory/src/com/pizzaIngredientFactory/PizzaIngredientFactory.java
package com.pizzaIngredientFactory;

import com.cheese.Cheese;
import com.clams.Clams;
import com.dough.Dough;
import com.pepperoni.Pepperoni;
import com.sauce.Sauce;
import com.veggies.Veggies;

public interface PizzaIngredientFactory {
 
	public Dough createDough();
	public Sauce createSauce();
	public Cheese createCheese();
	public Veggies[] createVeggies();
	public Pepperoni createPepperoni();
	public Clams createClam();
 
}

File: python/2.7/factory/com/pizza/pizzaIngredientFactory/PizzaIngredientFactory.py
from abc import ABCMeta, abstractmethod

class PizzaIngredientFactory:
    __metaclass__ = ABCMeta

    @abstractmethod
    def createDough(self):
        pass

    def createSauce(self):
        pass

    def createCheese(self):
        pass

    def createVeggies(self):
        pass

    def createPepperoni(self):
        pass

    def createClam(self):
        pass